embassy-stm32/src/pka/mod.rs

```diff
@@ -0,0 +1,503 @@
+//! PKA Hardware accelerator
+use core::future::poll_fn;
+use core::marker::PhantomData;
+use core::task::Poll;
+use core::{fmt, usize};
+
+use bytemuck::cast_slice;
+use embassy_hal_internal::{into_ref, Peripheral, PeripheralRef};
+use embassy_sync::waitqueue::AtomicWaker;
+
+use crate::interrupt::typelevel::Interrupt;
+use crate::{interrupt, pac, peripherals, rcc};
+
+static PKA_WAKER: AtomicWaker = AtomicWaker::new();
+
+// List of RAM addresses for data. Since embassy implementation of RAM access already moves base address to
+// 0x0400 and multiplies offset by 4 for 32-bit access, we have to take address from reference manual, subtract 0x0400
+// and divide it by 4 to get the correct offset.
+const OPERAND_LENGTH_ADDR: usize = 0x0001; // 0x0404
+const MODULUS_ADDR: usize = 0x0257; // 0x0D5C
+const MONT_PARAM_R2_MOD_N: usize = 0x0065; // 0x0594
+const OPERAND_A_ADDR: usize = 0x012D; // 0x08B4
+const OPERAND_B_ADDR: usize = 0x0191; // 0x0A44
+const ARITHMETIC_RESULT_ADDR: usize = 0x01F4; // 0x0BD0
+const EXPONENT_LENGTH_ADDR: usize = 0x0000; // 0x0400
+const EXPONENT_ADDR: usize = ARITHMETIC_RESULT_ADDR; // 0x0BD0
+const EXPONENTIATION_RESULT_ADDR: usize = 0x00C9; // 0x0724
```

Comment:
I could, but I would rather include more consts here to make it more readable. Some addresses in documentation share name, but not the address, i.e.: "Operand A" in Modular addition operation has address `0x8B4`, but "Operand A" in Modular exponentiation is located in `0xA44`, which is the address of "Operand B" in Arithmetic Subtraction or "Modulus value n" in Modular Inversion.

Reply:
I proposed an Enum because I thought they are more readable :laughing:  I think you can have same values for different names when you use Enum with `repr(u32)`, or you can have separate address Enum for particular usage. 

Treat this message as a suggestion, I'm not insisting that it should be done this way. I assume you know better what you're doing here, so I am marking this thread as resolved.

Reply:
I am good with embedded systems, but I still have long way to go wit "Rust way" :). I.e., I didn't know you could have multiple enums with the same value. That's convenient!

Reply:
I verified that, you can not have duplicated values in same enum, but mby you can define each address group as separate enum.